src/golibmc.go

```diff
@@ -921,14 +1325,69 @@ func (client *Client) Stats() (map[string](map[string]string), error) {
 
 // Quit will close the sockets to each memcached server
 func (client *Client) Quit() error {
-	client.lock()
-	defer client.unlock()
+	client.lk.Lock()
 
-	errCode := C.client_quit(client._imp)
-	C.client_destroy_broadcast_result(client._imp)
+	close(client.openerCh)
+	if client.cleanerCh != nil {
+		close(client.cleanerCh)
+	}
+	for _, cr := range client.connRequests {
+		close(cr)
+	}
+	client.closed = true
+	client.lk.Unlock()
+	for _, cn := range client.freeConns {
+		if err := cn.quit(); err != nil {
+			return err
+		}
+	}
+	client.lk.Lock()
+	client.freeConns = nil
+	client.lk.Unlock()
+
+	return nil
+}
 
-	if errCode == C.RET_CONN_POLL_ERR || errCode == C.RET_RECV_ERR {
+func (cn *conn) expired(timeout time.Duration) bool {
+	if timeout <= 0 {
+		return false
+	}
+	return cn.createdAt.Add(timeout).Before(time.Now())
+}
+
+func (cn *conn) quit() error {
```

Comment:
1. since quit is called internal and  there is nothing todo but logging, maybe quit can log inside and not return err?
2.  numOpen should always -- ?

Reply:
1. Some use err in upstream processing like [this](https://github.com/jumpeiMano/libmc/blob/c1060ece6e88dfe094348438714e2a1084809d42/src/golibmc.go#L1344-L1346)..
1. Fixed, thanks.
https://github.com/douban/libmc/pull/57/commits/ff59967e7cd3025256ed571af88e584feb68aedb